src/lib/features/snackbar/specs/snackbar.spec.ts

```diff
@@ -0,0 +1,39 @@
+import { render } from '@testing-library/svelte';
+import userEvent from '@testing-library/user-event';
+import { SnackbarQueue, snackbar, type SnackbarOptions } from '..';
+
+const firstSnackbar: SnackbarOptions = {
+  text: 'Hello',
+  buttonText: 'Close',
+  callback: jest.fn(),
+};
+
+const secondSnackbar: SnackbarOptions = {
+  text: 'Example',
+  buttonText: 'Close',
+  callback: jest.fn(),
+};
+
+it('adds a snackbar to the queue and displays it', () => {
+  snackbar(firstSnackbar);
+  const { getByText } = render(SnackbarQueue);
+  expect(getByText(firstSnackbar.text)).toBeInTheDocument();
+});
+
+it('adds two snackbars to the queue, but only one is displayed', () => {
+  snackbar(firstSnackbar);
+  snackbar(secondSnackbar);
+  const { queryByText } = render(SnackbarQueue);
+  expect(queryByText(firstSnackbar.text)).toBeInTheDocument();
+  expect(queryByText(secondSnackbar.text)).not.toBeInTheDocument();
+});
+
+it('removes snackbar from the queue after pressing the button', async () => {
+  const user = userEvent.setup();
+  snackbar(firstSnackbar);
+  const { getByRole, queryByText } = render(SnackbarQueue);
+
+  expect(queryByText(firstSnackbar.text)).toBeInTheDocument();
+  await user.click(getByRole('button', { name: firstSnackbar.buttonText }));
```

Comment:
suggestion: I think it's a good idea to also assert here that the callback was called